synapse_invite_checker/invite_checker.py

```diff
@@ -362,23 +434,33 @@ async def user_may_invite(
             # local invites are always valid, if they are not to a dm.
             # There are no checks done for outgoing invites apart from in the federation proxy,
             # which checks all requests.
-            # if self.api.is_mine(invitee):
             logger.debug("Local invite from %s to %s allowed", inviter, invitee)
             return NOT_SPAM
 
-        # Step 1, check federation allow list
         inviter_id = UserID.from_string(inviter)
         inviter_domain = inviter_id.domain
-        domain_set, _ = await self.fetch_federation_allow_list()
-        if inviter_domain not in domain_set:
-            self.fetch_federation_allow_list.cache_clear()
-            domain_set, _ = await self.fetch_federation_allow_list()
 
-            if inviter_domain not in domain_set:
-                logger.warning("Discarding invite from domain: %s", inviter_domain)
-                return errors.Codes.FORBIDDEN
+        # Step 1a, check federation allow list
+        if not await self.is_domain_allowed(inviter_domain):
+            logger.warning("Discarding invite from domain: %s", inviter_domain)
+            return errors.Codes.FORBIDDEN
+
+        # Step 1b, check that both domains are not of type 'epa'
+        # TODO: stop trusting our local setting and instead check the federation list
+        #  data to see if *this* server is of type `isInsurance`
```

Comment:
Is this a todo? I thought we intentionally use the epa setting?

```suggestion
        # We warn during startup if the 'epa' setting and isInsurance are not equivalent, so we can use the setting to know, if we are an insurance.
```

Reply:
this seems to be repeated later, so you can probably merge those comments.